module-4/backend/week_two.md

```diff
@@ -1,22 +1,71 @@
 ## Week Two - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. What's one difference between ES5 and ES6?
-2. What's the difference between asynchronous and synchronous JavaScript? 
+
+>ES6 has arrow funciton with implicit returns and ES5 does not
+
+2. What's the difference between asynchronous and synchronous JavaScript?
+
+>Synchronus, operates each function one at a time. Async fires all the functions
+at once. Only handling the results of each function once the result is available.
+
 3. What are the four pillars of Object Oriented programming?
+
+>Encapsulation, Abstraction, Polymorphism, Inheritance
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+
+>classes with constructor functions, libraries that we can inherit from to use already written code.
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+
+>I'm not sure I understand the question. I would assume some of concepts that need to
+be taken into consideration for refactoring would be the usualy code smells. Long conditional blocks,
+long functions, long parameter list, commented out code...
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+
+>Callback function is when you call a function with the (). It denotes a function that you want to call in a certain place or order.
```

Comment:
Not exactly. When we call a function we use the parenthesis. Callback functions are different, these are passed to other functions. The other function will then call that callback function. We have used these with event handlers and enumerable prototype methods